File: DataStructure_Alogorithm/Algorithm/Inflearn_basic_cpp/28_N에서0의개수/main.cpp
#include<stdio.h>
#include<algorithm>
#include<vector>
using namespace std;

int main() {
	int n,i,j,tmp, cntOne, cntTwo;

	scanf("%d", &n);

	for (i=2; i <= n; i++) {
		tmp = i;
		j = 2;
		while (1) {
			if (tmp % j == 0) {
				if (j == 2) cntOne++;
				else if (j == 5) cntTwo++;
				tmp = tmp / j;
			} 
			else j++;
			if (tmp ==1) break;
		}
	}

	if (cntOne > cntTwo) {
		printf("%d", cntOne);
	} else printf("%d", cntTwo);

	return 0;

}

File: DataStructure_Alogorithm/Algorithm/Inflearn_basic_cpp/28_N에서0의개수/main.go
package main

import (
	"fmt"
	"os"
)

func main() {
	file, _ := os.Open("input.txt")
	defer file.Close()
	old := os.Stdin
	defer func() { os.Stdin = old }()
	os.Stdin = file
	var N, twoCnt, fiveCnt int
	fmt.Scanf("%d", &N)

	for i := 2; i <= N; i++ {
		tmp := i
		j := 2
		for tmp > 1 {
			if tmp%j == 0 {
				if j == 2 {
					twoCnt++
				}
				if j == 5 {
					fiveCnt++
				}
				tmp = tmp / j
			} else {
				j++
			}
		}
	}

	if twoCnt > fiveCnt {
		fmt.Println(fiveCnt)
	} else {
		fmt.Println(twoCnt)
	}
}
